### PR TITLE
Implement Min/Max BSI queries

### LINKED ISSUE #909: Min Max in BSI
### Description
Support Min and MAX PQL operations for BSI fields
```
Min([BITMAP_CALL], <frame=STRING>, <field=STRING>)
Max([BITMAP_CALL], <frame=STRING>, <field=STRING>)
```

For BSI fields only this PQL operation would return the minimum value for that field and the bitmaps that contain that value.
This addition will make Pilosa more compatible with SPARQL. 

### Success criteria (What criteria will consider this ticket closeable?)
```
Return `{"min":7, "attrs":{},"bits":[30]}`
```

### PR DESCRIPTION
**Pull request by travisturner**
## Overview
Implements `Min` and `Max` functions on BSI fields.
I think this may require some discussion on the best way to handle `NaN` return values for the cases where either the frame is empty, or the filter results in zero columns being considered.

A couple of examples from the tutorial data:
```
Min(frame="measurements", field="age")
{"results":[{"min":19,"count":1}]}

Min(Range(frame="measurements", age > 30), frame="measurements", field="age")
{"results":[{"min":32,"count":1}]}

Max(Range(frame="measurements", age <= 2), frame="measurements", field="age")
{"results":[{"max":0,"count":0}]}
```

TODO:
- [x] Add some executor tests
- [x] Update the docs

Fixes #909

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: fragment_test.go [hunk 1]
```diff
@@ -256,6 +256,79 @@ func TestFragment_FieldSum(t *testing.T) {
...
+	}
+
+	t.Run("Min", func(t *testing.T) {
+		tests := []struct {
+			filter *pilosa.Bitmap
+			exp    uint64
+			cnt    uint64
+		}{
+			{filter: nil, exp: 0, cnt: 1},
+			{filter: pilosa.NewBitmap(2000, 4000, 5000), exp: 300, cnt: 2},
```
Comment (alanbernstein) [inline fragment_test.go]: I don't expect problems now, but maybe these values should include some negative numbers.

File: fragment_test.go [hunk 2]
```diff
@@ -256,6 +256,79 @@ func TestFragment_FieldSum(t *testing.T) {
...
+// Ensure a fragment can find the max of field values.
+func TestFragment_FieldMinMax(t *testing.T) {
+	const bitDepth = 16
+
+	f := test.MustOpenFragment("i", "f", pilosa.ViewStandard, 0, "")
+	defer f.Close()
+
+	// Set values.
+	if _, err := f.SetFieldValue(1000, bitDepth, 382); err != nil {
+		t.Fatal(err)
```
Comment (alanbernstein) [inline fragment_test.go]: I don't expect problems now, but maybe these values should include some negative numbers.

Comment (travisturner) [reply fragment_test.go]: @alanbernstein there are no negative values at the fragment level; that offset it applied by the frame. I added a negative number to the test in `executor.go`: 47a5ed84bd70e78d480b6d308ba2589c044b3c81

**Comment by yuce** [comment]
@travisturner This PR doesn't seem to have an protobuf definition for the min/max response.

**Comment by travisturner** [comment]
@yuce good point. 

I suppose at the very least we should change `SumCount` to something less specific (like `ValCount`), and support two additional types in the QueryResult.
Either:
```
message QueryResult {
    uint32 Type = 6;
    Bitmap Bitmap = 1;
    uint64 N = 2;
    repeated Pair Pairs = 3;
    ValCount ValCount = 5;
    bool Changed = 4;
}
```
or:
```
message QueryResult {
    uint32 Type = 6;
    Bitmap Bitmap = 1;
    uint64 N = 2;
    repeated Pair Pairs = 3;
    ValCount MinCount = 7;
    ValCount MaxCount = 8;
    ValCount SumCount = 5;
    bool Changed = 4;
}
```
What are your thoughts?

**Comment by yuce** [comment]
I think the first option is better for the moment.

**Comment by tgruben** [comment]
I vote for option 1 as well

**Comment by travisturner** [comment]
@yuce the latest commit changes `SumCount` to `ValCount` and uses that type for `Sum`, `Min`, and `Max`. Does that work for you?

**Comment by yuce** [comment]
@travisturner All tests pass in the Go client even if I didn't update the protobuf defs. I guess that's because the prev def has the same data layout with the current one.

**Pull request merged**